File: lab3/src/Vector.java
/**
 * Created by mamaxropela on 01.10.14.
 */
public class Vector {
    private int[] vector;

    public Vector(int size){
        vector = new int[size];
    }

    public void Generate(int filledDigit){
        for (int i = 0; i < vector.length; i++)
            vector[i] = filledDigit;
    }
    public int getElement(int index){
        return vector[index];
    }
    public void setElement(int index, int value){
        vector[index] = value;
    }

    public static Vector Multiply(Matrix matrix, Vector vector){
        Vector rezult = new Vector(vector.getSize());
        int tempValue = 0;
        for(int i = 0; i < rezult.getSize(); i++){
            tempValue = 0;
            for (int j = 0; j < rezult.getSize(); j++)
                tempValue += vector.getElement(i) * matrix.getElement(j,i);
            rezult.setElement(i,tempValue);
        }
        return rezult;
    }

    public static int Multiply(Vector v1, Vector v2){
        int rezult = 0;
        for (int i = 0; i < v1.getSize(); i++)
            rezult += v1.getElement(i)*v2.getElement(i);
        return rezult;
    }

    public static Vector Plus(Vector v1, Vector v2){
        Vector rezult = new Vector(v1.getSize());
        for (int i = 0; i < v1.getSize(); i++)
            rezult.setElement(i,v1.getElement(i) + v2.getElement(i));
        return rezult;
    }

    public int getSize(){
        return vector.length;
    }
}

File: 1 half/Lab5/lab5/Vector.cpp
#include "Data.h"
#include <iostream>

Vector::Vector(int sizes){
	size = sizes;
	vector = new int[size];
}

void Vector::generate(int filledDigit){
	for (int i = 0; i < size; i++)
		vector[i] = filledDigit;
}
int Vector::getElement(int index){
	return vector[index];
}

void Vector::setElement(int index, int value){
	vector[index] = value;
}

void Vector::print(){
	for (int i = 0; i < size; i++)
		std::printf("%6d  ", vector[i]);
	std::printf("\n");
}

int Vector::operator*(Vector &vec){
	int rezult = 0;
	for (int i = 0; i < size; i++)
		rezult += vector[i]*vec.getElement(i);
	return rezult;
}

Vector* Vector::operator+(Vector &vec){
	Vector *rezult = new Vector(size);
	for (int i = 0; i < size; i++)
		rezult->setElement(i, vector[i] + vec.getElement(i));
	return rezult;
}

Vector* Vector::sort(){
	Vector * rezult = new Vector(size);
	std::copy(vector, (int*)vector + size, rezult->vector);
	for (int i = 0; i < size; i++)
		for (int j = 0; j < size - i - 1; j++)
			if (rezult->vector[j]>rezult->vector[j + 1])
				std::swap(rezult->vector[j], rezult->vector[j + 1]);
	return rezult;
}

int Vector::getSize(){
	return size;
}

Vector::~Vector()
{
	delete[] vector;
}
